near-contract-standards/src/event.rs

```diff
@@ -0,0 +1,264 @@
+use near_sdk::AccountId;
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+
+#[derive(Serialize, Debug)]
+#[serde(tag = "standard")]
+#[serde(rename_all = "snake_case")]
+pub enum NearEvent<'a> {
+    #[serde(borrow)]
+    Nep171(Nep171Event<'a>),
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Nep171Event<'a> {
+    pub version: &'static str,
+    #[serde(flatten)]
+    #[serde(borrow)]
+    pub event_kind: Nep171EventKind<'a>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "event", content = "data")]
+#[serde(rename_all = "snake_case")]
+#[allow(clippy::enum_variant_names)]
+pub enum Nep171EventKind<'a> {
+    #[serde(borrow)]
+    NftMint(Vec<NftMintData<'a>>),
+    #[serde(borrow)]
+    NftTransfer(Vec<NftTransferData<'a>>),
+    #[serde(borrow)]
+    NftBurn(Vec<NftBurnData<'a>>),
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftMintData<'a> {
+    #[serde(borrow)]
+    pub owner_id: &'a str,
+    #[serde(borrow)]
+    pub token_ids: Vec<&'a str>,
+    #[serde(borrow)]
+    pub memo: Option<&'a str>,
+}
+
+impl<'a> NftMintData<'a> {
+    pub fn new(
+        owner_id: &'a AccountId,
+        token_ids: Vec<&'a str>,
+        memo: Option<&'a str>,
+    ) -> NftMintData<'a> {
+        Self { owner_id: owner_id.as_str(), token_ids, memo }
+    }
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NftTransferData<'a> {
+    #[serde(borrow)]
+    pub old_owner_id: &'a str,
+    #[serde(borrow)]
+    pub new_owner_id: &'a str,
```

Comment:
why not these as account ids?

Reply:
The AccountId references are used in the constructors.  If I add them here I get:

```rust
the trait bound `&'a AccountId: Deserialize<'_>` is not satisfied
the following implementations were found:
  <AccountId as Deserialize<'de>>
  ```

Reply:
yeah, the point before is that the deserialize isn't needed for anything. nbd can leave as is